hail/python/hail/backend/local_backend.py

```diff
@@ -219,22 +101,14 @@ def register_ir_function(self,
             return_type._parsable_string(),
             jbody)
 
-    def _is_registered_ir_function_name(self, name: str) -> bool:
-        return name in self._registered_ir_function_names
-
     def stop(self):
+        self._backend_server.stop()
```

Comment:
You need to stop this in the SparkBackend as well. I think it really should live on the Py4JBackend and both subclasses should call their super methods.